Added climate.modbus module documentation

Kirchoff opened the pull request:
**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#12224
## Checklist:

- [x] Branch: Fixes, changes and adjustments should be created against `current`. New documentation for platforms/components and features should go to `next`.
- [x] The documentation follow the [standards][standards].

Commit:
Added climate.modbus module documentation

Kirchoff committed:
Update climate.modbus.markdown

Updated IOT class Local Push to Local Polling.

dgomes commented:
Can you provide more information about using this component ? E.g. What equipments works with this ? Does it require any extra hardware ?

frenck commented on source/_components/climate.modbus.markdown:
````diff
@@ -0,0 +1,52 @@
+
+```
+
+Configuration variables:
````
Please use the configuration tags, for more information please see:
https://home-assistant.io/developers/documentation/create_page/#configuration

frenck commented on source/_components/climate.modbus.markdown:
```diff
@@ -0,0 +1,52 @@
+    target_temp_register: 2782
+    # Universal Input 1 Module 1
+    current_temp_register: 27586
+    data_type: float
```
Please don't add optional configuration parameters to the examples.
We'd like to keep examples as minimal as possible.

frenck commented on source/_components/climate.modbus.markdown:
```diff
@@ -0,0 +1,52 @@
+footer: true
+logo: modbus.png
+ha_category: Thermostat
+ha_release: pre 0.7
```
This is a new component. Please set the `ha_release` to the next release (`0.66`).

frenck reviewed (changes requested):
@Kirchoff Thank you for this. Left a couple of small RFC's. Could you please take a look? 👍

Kirchoff committed:
Reviewed comments for changes

Added configuration tags, removed optional configuration values, and set the correct release version.

Kirchoff commented:
@frenck - I made the changes you requested, does everything look good? Thank you.

frenck commented on source/_components/climate.modbus.markdown:
````diff
@@ -0,0 +1,48 @@
+```
+
+{% configuration %}
+  - **name** (*Required*): Name of the sensor.
````
Almost correct! You need to change/format the configuration options as well.

https://home-assistant.io/developers/documentation/create_page/#configuration

frenck reviewed (changes requested):
@Kirchoff Almost good!

Commit:
Fixed formatting with the configuration settings

frenck commented:
Rebased the PR in order to fix the build.

frenck reviewed (dismissed):
Thank you @Kirchoff! 🏅 

Can be merged as soon as the parent PR gets merged.

frenck committed:
:arrow_up: ha_release -> 0.68
